service/worker/deployment/deployment_workflow.go

```diff
@@ -115,7 +118,10 @@ func (d *DeploymentWorkflowRunner) run() error {
 			if d.DeploymentLocalState.TaskQueueFamilies == nil {
 				d.DeploymentLocalState.TaskQueueFamilies = make(map[string]*deployspb.DeploymentLocalState_TaskQueueFamilyInfo)
 			}
-			// if no TaskQueues have been registered for the TaskQueueName
+
+			if _, ok := d.DeploymentLocalState.TaskQueueFamilies[updateInput.TaskQueueName]; !ok {
+				d.DeploymentLocalState.TaskQueueFamilies[updateInput.TaskQueueName] = &deployspb.DeploymentLocalState_TaskQueueFamilyInfo{}
+			}
 			if d.DeploymentLocalState.TaskQueueFamilies[updateInput.TaskQueueName].TaskQueues == nil {
 				d.DeploymentLocalState.TaskQueueFamilies[updateInput.TaskQueueName].TaskQueues = make(map[int32]*deployspb.DeploymentLocalState_TaskQueueFamilyInfo_TaskQueueInfo)
 			}
```

Comment:
the changes in this file were cherry-picked from https://github.com/temporalio/temporal/pull/6864/commits/8c856b829c38f073f6056d24022cea8f74da9845 to get the server to build